CMakeLists.txt

```
@@ -0,0 +1,225 @@
+        ${M_LIBRARIES})
+endif()
+
+target_link_libraries(mold PRIVATE
+    TBB::tbb
+    $<$<BOOL:MOLD_USE_MIMALLOC>:mimalloc::mimalloc>)
+
+if(CMAKE_SYSTEM_NAME MATCHES Linux)
+    find_package(OpenSSL REQUIRED COMPONENTS Crypto)
+    target_link_libraries(mold PRIVATE OpenSSL::Crypto)
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES riscv64
+    OR CMAKE_SYSTEM_PROCESSOR MATCHES armv6)
+    find_library(ATOMIC_LIBRARIES atomic REQUIRED)
+    target_link_libraries(mold PRIVATE ${ATOMIC_LIBRARIES})
+endif()
+
+file(STRINGS .git/HEAD GIT_HASH)
+
+if(GIT_HASH MATCHES "ref: ")
+    string(REPLACE "ref: " "" GIT_HASH "${GIT_HASH}")
+    file(STRINGS .git/${GIT_HASH} GIT_HASH)
+endif()
+
+target_compile_definitions(mold PRIVATE
+    GIT_HASH="${GIT_HASH}"
+    MOLD_VERSION="${CMAKE_PROJECT_VERSION}")
+
+target_sources(mold PRIVATE
+    compress.cc
+    demangle.cc
+    filepath.cc
+    glob.cc
+    hyperloglog.cc
+    main.cc
+    multi-glob.cc
+    perf.cc
+    strerror.cc
+    tar.cc
+    uuid.cc
+    elf/arch-arm32.cc
+    elf/arch-arm64.cc
+    elf/arch-i386.cc
+    elf/arch-riscv64.cc
+    elf/arch-x86-64.cc
+    elf/cmdline.cc
+    elf/dwarf.cc
+    elf/gc-sections.cc
+    elf/icf.cc
+    elf/input-files.cc
+    elf/input-sections.cc
+    elf/linker-script.cc
+    elf/lto.cc
+    elf/main.cc
+    elf/mapfile.cc
+    elf/output-chunks.cc
+    elf/passes.cc
+    elf/relocatable.cc
+    elf/subprocess.cc
+    macho/arch-arm64.cc
+    macho/arch-x86-64.cc
+    macho/cmdline.cc
+    macho/dead-strip.cc
+    macho/input-files.cc
+    macho/input-sections.cc
+    macho/lto.cc
+    macho/main.cc
+    macho/mapfile.cc
+    macho/output-chunks.cc
+    macho/tapi.cc
+    macho/yaml.cc
+    third-party/rust-demangle/rust-demangle.c)
+
+# Create the ld and ld64 symlinks required for testing
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld symlink"
+    VERBATIM)
+add_custom_command(
+    TARGET mold POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E create_symlink mold ld64
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Creating ld64 symlink"
+    VERBATIM)
+
+include(CTest)
+
+set(MOLD_TESTS_PATH ${CMAKE_CURRENT_LIST_DIR}/test/)
+
+if(CMAKE_SYSTEM_NAME MATCHES Darwin)
+    string(APPEND MOLD_TESTS_PATH "macho")
+else()
+    string(APPEND MOLD_TESTS_PATH "elf")
+endif()
+
+string(APPEND MOLD_TESTS_PATH "/*")
+FILE(GLOB MOLD_TESTS ${MOLD_TESTS_PATH})
+
+foreach(MOLD_TEST IN LISTS MOLD_TESTS)
+    add_test(NAME ${MOLD_TEST} COMMAND ${MOLD_TEST})
+endforeach()
+
+# FIXME(cwasser): This is just a convenience ctest target,
+# which ensures that mold is build. This should ideally be fixed in CMake.
+include(ProcessorCount)
+ProcessorCount(MOLD_NPROC)
+add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} -j${MOLD_NPROC}
+    DEPENDS mold USES_TERMINAL)
+
+install(TARGETS mold DESTINATION ${CMAKE_INSTALL_BINDIR})
```

```suggestion
install(TARGETS mold)
```

CMake >= 3.14 will fill the destinations properly if omitted for a few groups, `RUNTIME` being one of those.

---

Thanks!